static/app/views/performance/newTraceDetails/traceDrawer/traceDrawer.tsx

```diff
@@ -0,0 +1,201 @@
+};
+
+const MIN_PANEL_HEIGHT = 100;
+const INITIAL_PANEL_HEIGHT = 200;
+
+function getNodeTabTitle(node: TraceTreeNode<TraceTree.NodeValue>) {
+  if (isTransactionNode(node)) {
+    return node.value['transaction.op'] + ' - ' + node.value.transaction;
+  }
+
+  if (isSpanNode(node)) {
+    return node.value.op + ' - ' + node.value.description;
+  }
+
+  if (isAutogroupedNode(node)) {
+    return t('Auto-Group');
+  }
+
+  if (isMissingInstrumentationNode(node)) {
+    return t('Missing Instrumentation Span');
+  }
+
+  if (isTraceErrorNode(node)) {
+    return node.value.title;
+  }
+
+  return t('Detail');
+}
+
+function TraceDrawer(props: DrawerProps) {
+  const [size, setSize] = useState(INITIAL_PANEL_HEIGHT);
+
+  const [isResizing, setIsResizing] = useState(false);
+
+  const panelRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const handleMouseMove = e => {
+      if (!isResizing) return;
+      const newSize = Math.max(MIN_PANEL_HEIGHT, size + e.movementY * -1);
+      setSize(newSize);
+    };
+
+    const handleMouseUp = () => setIsResizing(false);
+
+    document.addEventListener('mousemove', handleMouseMove);
+    document.addEventListener('mouseup', handleMouseUp);
+    return () => {
+      document.removeEventListener('mousemove', handleMouseMove);
+      document.removeEventListener('mouseup', handleMouseUp);
+    };
+  }, [size, isResizing]);
+
+  useEffect(() => {
+    props.setDetailPanelRef(panelRef);
+  }, [panelRef, props]);
+
+  const handleMouseDown = e => {
+    e.preventDefault();
+    setIsResizing(true);
+  };
+
+  return (
+    <PanelWrapper size={size} ref={panelRef}>
+      <TabsContainer onMouseDown={handleMouseDown}>
+        {props.nodes.map((node, index) => (
+          <Tooltip title={getNodeTabTitle(node)} showOnlyOnOverflow key={index}>
+            <Tab
+              key={index}
+              active={props.activeTab === 'node_detail'}
+              onClick={() => props.setActiveTab('node_detail')}
+            >
+              {getNodeTabTitle(node)}
+            </Tab>
+          </Tooltip>
+        ))}
+        <Tab
+          active={props.activeTab === 'trace_data'}
+          onClick={() => props.setActiveTab('trace_data')}
+        >
+          {t('Trace')}
+        </Tab>
+      </TabsContainer>
+
+      <Content>
+        {props.activeTab === 'trace_data' && (
+          <TraceLevelDetails
+            rootEventResults={props.rootEventResults}
+            organization={props.organization}
+            location={props.location}
+            traces={props.traces}
+            traceEventView={props.traceEventView}
+          />
+        )}
+        {props.activeTab === 'node_detail' &&
+          props.nodes.map((node, index) => (
+            <NodeDetail
+              key={index}
+              node={node}
+              organization={props.organization}
+              location={props.location}
+            />
+          ))}
```

Comment:
This should be something like
props.tab === trace_data return <Trace />
props.tab === node_detail <Trace />
throw error